platform/openide.loaders/src/org/openide/loaders/FolderInstance.java

```diff
@@ -115,14 +115,14 @@
     /** Array of tasks that we have to check before we are ok. These are the tasks
      *  associated with children of the current folder.
      */
-    private Task[] waitFor;
+    private volatile Task[] waitFor;
```

Comment:
Volatile applies only to reading the array pointer, but not to accessing the elements of the array. Is that OK?

Reply:
It's a pity we can not make `folder` final, since it's API.

Reply:
In addition to the original change fields `container`, `map`, `listener` and `err` are now final, field `creationTask` is volatile.